app/models/concerns/email.rb

```diff
@@ -0,0 +1,19 @@
+# Common behaviour for models that store email addresses
+module Email
+  extend ActiveSupport::Concern
+
+  included do
+    validates :email, presence: true, uniqueness: true, case_sensitive: false
+    validates_with EmailAddress::ActiveRecordValidator
+    validates :canonical_email, presence: true
+    validates_with EmailAddress::ActiveRecordValidator
```

Comment:
Lines 7 and 9 are repeated, it looks like they should make use of the `field` or `fields` option to specify which attribute to validate: https://github.com/afair/email_address#rails-validator